src/functional/ffront/ast_passes/single_static_assign.py

```diff
@@ -17,13 +17,23 @@
 from functional.ffront.fbuiltins import TYPE_BUILTIN_NAMES
 
 
+def _make_assign(target: str, source: str, location_node: ast.AST):
+    result = ast.Assign(
+        targets=[ast.Name(ctx=ast.Store(), id=target)], value=ast.Name(ctx=ast.Load(), id=source)
+    )
+    for node in ast.walk(result):
+        ast.copy_location(node, location_node)
+    return result
+
+
 class SingleStaticAssignPass(ast.NodeTransformer):
     """
     Rename variables in assignments to avoid overwriting.
 
     Mutates the python AST, variable names will not be valid python names anymore.
     This pass must be run before any passes that linearize unpacking assignments.
 
+    Requires AST to have no missing location info (see ``FixMissingLocations``).
```

Comment:
Is this really true? If some location info is missing `ast.copy_location` will just do nothing.

Reply:
I have a faint memory that it was needed for something, but I think my mind is in the wrong here :sweat_smile: 

Will remove the comment.

Reply:
Removed